docs/settings.rst
.. _ref-settings:

==========================
Configuration and defaults
==========================

Gum options
===========

GUM_DEBUG
---------

Activates/deactivates the debug output.

Defaults to ``False``.

GUM_USE_CELERY
--------------

Boolean setting to activate/desactivate the use of Celery
to update a model document when it is saved.

Defaults to ``False``.

GUM_INDEX_FILES
---------------

You can use this variable to give a list of index files to be checked
by Gum in order to find registered mappings, in addition to index.py files.

Defaults to ``tuple()``.

Elasticsearch options
=====================

GUM_ELASTICSEARCH_URLS
----------------------

A list of address of Elasticsearch servers.

Defaults to ``["http://127.0.0.1:9200/"]``.

GUM_ELASTICSEARCH_CONNECTION_PARAMS
-----------------------------------

Parameters given to `Elasticsearch class <https://elasticsearch-py.readthedocs.org/en/master/api.html#elasticsearch.Elasticsearch>`_ to
connect with Elasticsearch server.

Defatils to::

    {
        "timeout": 5
    }

GUM_ELASTICSEARCH_INDEX
-----------------------

Index name to use by default.

Defaults to ``"_all"``.

GUM_DEFAULT_ELASTICSEARCH_SETTINGS
----------------------------------

Defines the settings of all indices created by Gum.

Defaults to::

    {
        'settings': {
            "analysis": {
                "analyzer": {
                    "ngram_analyzer": {
                        "type": "custom",
                        "tokenizer": "lowercase",
                        "filter": ["gum_ngram"]
                    },
                    "edgengram_analyzer": {
                        "type": "custom",
                        "tokenizer": "lowercase",
                        "filter": ["gum_edgengram"]
                    }
                },
                "tokenizer": {
                    "gum_ngram_tokenizer": {
                        "type": "nGram",
                        "min_gram": 3,
                        "max_gram": 15,
                    },
                    "gum_edgengram_tokenizer": {
                        "type": "edgeNGram",
                        "min_gram": 2,
                        "max_gram": 15,
                        "side": "front"
                    }
                },
                "filter": {
                    "gum_ngram": {
                        "type": "nGram",
                        "min_gram": 3,
                        "max_gram": 15
                    },
                    "gum_edgengram": {
                        "type": "edgeNGram",
                        "min_gram": 2,
                        "max_gram": 15
                    }
                }
            }
        }
    }


HISTORY.rst
.. :changelog:

History
-------

1.1.1 (2016-04-12)
++++++++++++++++++

* Fixed problem building connections cache key
* Updated documentation

1.1.0 (2016-04-05)
++++++++++++++++++

* Added ``urls`` attribute to ``MappingType`` to be able to custom urls of Elasticsearch servers
* Added ``GenericElasticsearchManager`` class to make queries without specific ``MappingType`` class


1.0.1 (2016-03-31)
++++++++++++++++++

* Fixed problem running --update-settings


1.0.0 (2016-03-24)
++++++++++++++++++

* New settings var for configure Elasticsearch connection.
* Better project structure for distribution.


1.0.0rc1 (2016-02-29)
+++++++++++++++++++++

* First release on PyPI.


gum/settings.py
# -*- coding: utf-8 -*-
from __future__ import unicode_literals
from django.conf import settings


# List of URLs of Elasticsearch servers
DEFAULT_ELASTICSEARCH_URLS = ["http://127.0.0.1:9200/"]
ELASTICSEARCH_URLS = getattr(settings, "GUM_ELASTICSEARCH_URLS", DEFAULT_ELASTICSEARCH_URLS)

# Params for connection to Elasticsearch
DEFAULT_ELASTICSEARCH_TIMEOUT = 5
ELASTICSEARCH_TIMEOUT = getattr(settings, "GUM_ELASTICSEARCH_TIMEOUT", DEFAULT_ELASTICSEARCH_TIMEOUT)
DEFAULT_ELASTICSEARCH_CONNECTION_PARAMS = {
    "timeout": ELASTICSEARCH_TIMEOUT
}
ELASTICSEARCH_CONNECTION_PARAMS = getattr(
    settings, "GUM_ELASTICSEARCH_CONNECTION_PARAMS", DEFAULT_ELASTICSEARCH_CONNECTION_PARAMS
)

# A comma-separated list of index names the alias should point to (supports wildcards);
# use _all or omit to perform the operation on all indices.
DEFAULT_INDICES = "_all"
ELASTICSEARCH_INDICES = getattr(settings, "GUM_ELASTICSEARCH_INDEX", DEFAULT_ELASTICSEARCH_TIMEOUT)

DEBUG = getattr(settings, 'GUM_DEBUG', False)
USE_CELERY = getattr(settings, 'GUM_USE_CELERY', False)
INDEX_FILES = tuple(getattr(settings, 'GUM_INDEX_FILES', ()))

# Default configuration for indices
DEFAULT_ELASTICSEARCH_SETTINGS = getattr(
    settings,
    'GUM_DEFAULT_ELASTICSEARCH_SETTINGS',
    {
        'settings': {
            "analysis": {
                "analyzer": {
                    "ngram_analyzer": {
                        "type": "custom",
                        "tokenizer": "lowercase",
                        "filter": ["gum_ngram"]
                    },
                    "edgengram_analyzer": {
                        "type": "custom",
                        "tokenizer": "lowercase",
                        "filter": ["gum_edgengram"]
                    }
                },
                "tokenizer": {
                    "gum_ngram_tokenizer": {
                        "type": "nGram",
                        "min_gram": 3,
                        "max_gram": 15,
                    },
                    "gum_edgengram_tokenizer": {
                        "type": "edgeNGram",
                        "min_gram": 2,
                        "max_gram": 15,
                        "side": "front"
                    }
                },
                "filter": {
                    "gum_ngram": {
                        "type": "nGram",
                        "min_gram": 3,
                        "max_gram": 15
                    },
                    "gum_edgengram": {
                        "type": "edgeNGram",
                        "min_gram": 2,
                        "max_gram": 15
                    }
                }
            }
        }
    }
)
